collectors/cronjob.go

```diff
@@ -17,16 +17,22 @@ limitations under the License.
 package collectors
 
 import (
+	"time"
+
 	"github.com/golang/glog"
 	"github.com/prometheus/client_golang/prometheus"
+	"github.com/robfig/cron"
 	"golang.org/x/net/context"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/pkg/api"
 	v2batch "k8s.io/client-go/pkg/apis/batch/v2alpha1"
 	"k8s.io/client-go/tools/cache"
 )
 
+type nowFuncT func() time.Time
+
 var (
+	nowFunc nowFuncT // so that we can mock out `time.Now()` in tests
```

Comment:
This would usually be a plain assignment, `timeNow = time.Now` -- no need for the extra type or the wrapper in `init()`.

Alternatively, `now` could be passed to `getSchedulingDelaySeconds()` as a parameter and then we wouldn't need to stub it out at all.

(heh, you can take the reviewer out of go-readability-approvers@, but ...)